geth/node/manager.go

```diff
@@ -70,13 +76,91 @@ func (m *NodeManager) startNode(config *params.NodeConfig) (<-chan struct{}, err
 		return nil, ErrNodeExists
 	}
 
-	ethNode, err := MakeNode(config)
+	m.nodeSyncStarted = make(chan struct{}, 0)
+	m.nodeSyncError = make(chan struct{}, 0)
+	m.nodeSyncCompleted = make(chan struct{}, 0)
+
+	ethNode, err := MakeNode(config, func(leth *les.LightEthereum, stack *node.Node, config *params.NodeConfig) {
+		downlder := leth.Downloader()
+
+		go func() {
+			syncStateWait := time.NewTimer(params.DelayCycleForSyncStart)
+			syncStateWaitMax := time.NewTimer(params.DelayCycleForSyncStartMaxWait)
+
+			defer syncStateWait.Stop()
+			defer syncStateWaitMax.Stop()
+
+			// Initially we want to listen for when node have started synchronizing it's data.
+		nloop:
+			for {
+				select {
+				case <-m.nodeSyncError:
+					// Listen for error to ensure we are not in a case of concurrent runs of this,
+					// as I noticed an issue of double closure of channels, but no way of knowing why.
+					// This provides us a benefit that the go-routine will instantly be shutdown if the channel
+					// was already closed.
+					return
+				case <-syncStateWaitMax.C:
+					close(m.nodeSyncError)
```

Comment:
The channel are used as signals and not to encapsulate specific data, the `m.nodeSyncError` channel will be closed where we can then use `select` to use them to signal what the state of sync was, if the `m.nodeSyncError` was closed then the sync failed.

We could how ever include a buffer for a single item to contain but i think this is rather un-necessary, as we want to know status of sink. Unless we wish to had more context like:

`Sync still at block level` type errors, it would not be so necessary.

Reply:
I get the channels role and it's fine. Just wonder, if we want to store the sync error and provide a method to access it. We can skip this as I guess it's not needed.